examples/threathunter/threat_intelligence/stix_taxii.py

```diff
@@ -0,0 +1,270 @@
+    with open(config_filename, "r") as config_file:
+        config_data = yaml.load(config_file, Loader=yaml.SafeLoader)
+        logging.info(f"loaded config data: {config_data}")
+        return config_data
+
+
+@dataclass(eq=True, frozen=True)
+class TaxiiSiteConfig:
+    feed_id: str = ''
+    site: str = ''
+    discovery_path: str = ''
+    collection_management_path: str = ''
+    poll_path: str = ''
+    use_https: bool = False
+    ssl_verify: bool = False
+    cert_file: str = None
+    key_file: str = None
+    default_score: int = 5  # [1,10]
+    username: str = None
+    password: str = None
+    collections: str = '*'
+    start_date: str = '2019-01-01 00:00:00'
+    minutes_to_advance: int = 60
+    ca_cert: str = None
+    http_proxy_url: str = None
+    https_proxy_url: str = None
+    reports_limit: int = 10000
+    fail_limit: int = 10   # num attempts per collection for polling & parsing
+
+
+class TaxiiSiteConnector():
+    def __init__(self, site_conf):
+        self.config = TaxiiSiteConfig(**site_conf)
+        self.client = None
+
+    def create_taxii_client(self):
+        conf = self.config
+        if not conf.ssl_verify:
+            urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+        try:
+            client = create_client(conf.site,
+                                   use_https=conf.use_https,
+                                   discovery_path=conf.discovery_path)
+            client.set_auth(username=conf.username,
+                            password=conf.password,
+                            verify_ssl=conf.ssl_verify,
+                            ca_cert=conf.ca_cert,
+                            cert_file=conf.cert_file,
+                            key_file=conf.key_file)
+
+            proxy_dict = dict()
+            if conf.http_proxy_url:
+                proxy_dict['http'] = conf.http_proxy_url
+            if conf.https_proxy_url:
+                proxy_dict['https'] = conf.https_proxy_url
+            if proxy_dict:
+                client.set_proxies(proxy_dict)
+
+            self.client = client
+
+        except handled_exceptions as e:
+            logging.error(f"Error creating client: {e}")
+
+
+
+    def create_uri(self, config_path):
+        uri = None
+        if self.config.site and config_path:
+            if self.config.use_https:
+                uri = 'https://'
+            else:
+                uri = 'http://'
+            uri = uri + self.config.site + config_path
+        return uri
+
+
+
+    def query_collections(self):
+        collections = []
+        try:
+            uri = self.create_uri(self.config.collection_management_path)
+            collections = self.client.get_collections(
+                uri=uri)  # autodetect if uri=None
+            for collection in collections:
+                logging.debug(f"Collection: {collection.name}, {collection.type}")
```

Comment:
This is adding extra for-loop just to log out the collection, could a check be used to short-circuit this if the log level isn't set to debug?

Reply:
This is useful information for the user. I like the idea of short-circuiting, but I've changed this to an INFO logging statement to reflect the info to the user